src/App.js

```diff
@@ -25,7 +32,10 @@ function App() {
           element={token ? <Playlist /> : <Login />}
         ></Route>
         <Route path='/playlist/:playlistId' element={<PlaylistVideo />}></Route>
-        
+        <Route
+          path='/history'
+          element={token ? <History /> : <Login />}
```

Comment:
Rather than doing this multiple times, you could make a wrapper to redirect to Login component

Reply:
I will change it. thank you for your suggestion